discovery_engine_core/core/src/stack/exploration.rs

```diff
@@ -64,8 +64,16 @@ impl Stack {
             return Ok(());
         }
 
-        Self::validate_documents_stack_id(new_documents, Stack::id())?;
-        let documents = chain!(&self.data.documents, new_documents)
+        let new_documents = new_documents
+            .iter()
+            .cloned()
+            .map(|mut doc| {
+                doc.stack_id = Self::id();
+                doc
+            })
+            .collect_vec();
```

Comment:
:grey_question: Is `collect_vec` really needed? We should be able to chain this iterator with an iterator from `self.data.documents`, something like: `self.data.documents.iter().cloned().chain(new_documents).collect()`.

Reply:
This will also avoid to clone two times the documents from `new_documents`.

Reply:
Indeed your suggestion is better 👍
i adjusted it